src/Microsoft.OData.Edm/Csdl/Parsing/SchemaJsonParser.cs

```diff
@@ -0,0 +1,1564 @@
+//---------------------------------------------------------------------
+// <copyright file="SchemaJsonParser.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+#if NETSTANDARD2_0
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text.Json;
+using Microsoft.OData.Edm.Csdl.Parsing.Ast;
+using Microsoft.OData.Edm.Validation;
+using Microsoft.OData.Edm.Vocabularies.V1;
+
+namespace Microsoft.OData.Edm.Csdl.Parsing
+{
+    /// <summary>
+    /// Provides for the loading and conversion of one or more CSDL XML readers into Entity Data Model.
+    /// </summary>
+    internal static class SchemaJsonParser
+    {
+        /// <summary>
+        /// Parse the input <see cref="JsonElement"/> to the <see cref="CsdlSchema"/>.
+        /// </summary>
+        /// <param name="schemaNamespace">The namespace of the schema.</param>
+        /// <param name="element">The json element to parse from.</param>
+        /// <param name="context">The parser context.</param>
+        /// <returns>The parsed <see cref="CsdlSchema"/>.</returns>
+        internal static CsdlSchema ParseCsdlSchema(string schemaNamespace, Version version, JsonElement element, JsonParserContext context)
+        {
+            // A schema is represented as a member of the document object whose name is the schema namespace.
+            Debug.Assert(element.ValueKind == JsonValueKind.Object);
+
+            IList<CsdlAnnotations> outOfLineAnnotations = new List<CsdlAnnotations>();

```

Comment:
>IList [](start = 12, length = 5)

why IList?

Reply:
List<T> implements IList<T>, so it's ok to use base type reference.

---
In reply to: [443050374](https://github.com/OData/odata.net/pull/1775#discussion_r443050374) [](ancestors = 443050374)